maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/metadata/AddPluginArtifactMetadataMojo.java

```diff
@@ -18,10 +18,8 @@
  */
 package org.apache.maven.plugin.plugin.metadata;
 
-import org.apache.maven.artifact.Artifact;
 import org.apache.maven.artifact.repository.metadata.ArtifactRepositoryMetadata;
 import org.apache.maven.artifact.repository.metadata.GroupRepositoryMetadata;
```

Comment:
Those two imports should be removed, they are referenced from the javadoc and they should be moved to LegacySupport.